demo/src/main/java/com/demo/DemoAccessChecker.java

```diff
@@ -0,0 +1,39 @@
+package com.demo;
+
+import ca.uhn.fhir.context.FhirContext;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.google.fhir.gateway.HttpFhirClient;
+import com.google.fhir.gateway.interfaces.AccessChecker;
+import com.google.fhir.gateway.interfaces.AccessCheckerFactory;
+import com.google.fhir.gateway.interfaces.AccessDecision;
+import com.google.fhir.gateway.interfaces.NoOpAccessDecision;
+import com.google.fhir.gateway.interfaces.PatientFinder;
+import com.google.fhir.gateway.interfaces.RequestDetailsReader;
+import javax.inject.Named;
+
+/** Simple access checker to only allow patient list operation. */
+public class DemoAccessChecker implements AccessChecker {
+
+  DemoAccessChecker() {}
+
+  @Override
+  public AccessDecision checkAccess(RequestDetailsReader requestDetails) {
+    if (requestDetails.getRequestPath().equalsIgnoreCase("patient")) {
```

Comment:
Why ignore-case? We are expecting this to be a FHIR query so I think it is fair to only allow `/Patient`.